### PR TITLE
Roundstart species (Vatborn sprites II)

### PR DESCRIPTION
**Pull request by XSlayer300**
## About The Pull Request
Adds the ability to play as any species from roundstart, this is a sequel to #4868 where Vatborns are now a spawnable species. In addition to this, Vat-Growns (cloned ingame humans) now have their unique sprites instead of the boring SS13 human sprite we know. Though the only change between Vatborns and Humans are simply cosmetic, but they have differing naming formats and sprites.

## Why It's Good For The Game
More variety in the game, not to mention we have a bunch of people named "CS-F/M-'Codename'" nowadays.

## Changelog
:cl:
add: Roundstart species, allowing vatborns to be used in startup prefs.
/:cl:

### CODE REVIEW

File: code/__HELPERS/global_lists.dm [hunk 1]
```diff
@@ -54,6 +54,7 @@ GLOBAL_LIST_EMPTY(randomized_pill_icons)
 		var/datum/species/S = new T
 		S.race_key = rkey //Used in mob icon caching.
 		GLOB.all_species[S.name] = S
+		GLOB.roundstart_species[S.name] = S
```
Comment (SplinterGP) [inline code/__HELPERS/global_lists.dm:57]: make a var on species or a way to see if theyre joinable roundstart.
```suggestion
		if(S.joinable_roundstart)
			GLOB.roundstart_species[S.name] = S
```
in example.

File: code/_globalvars/lists/mobs.dm [hunk 2]
```diff
@@ -43,6 +43,7 @@ GLOBAL_LIST_EMPTY(language_datum_instances)
 GLOBAL_LIST_EMPTY(all_languages)
 
 GLOBAL_LIST_EMPTY(all_species)
+GLOBAL_LIST_INIT(roundstart_species, list("Human", "Vatborn"))
```
Comment (SplinterGP) [inline code/_globalvars/lists/mobs.dm:46]:
```suggestion
GLOBAL_LIST_EMPTY(roundstart_species)
```
Start this list empty as it is Filled like all_species.

File: code/modules/client/preferences.dm [hunk 3]
```diff
@@ -252,7 +252,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 	dat += "<br>"
 	dat += "<b>Eye:</b> <a href='?_src_=prefs;preference=eyecolor'>Color</a> <font face='fixedsys' size='3' color='#[num2hex(r_eyes, 2)][num2hex(g_eyes, 2)][num2hex(b_eyes, 2)]'><table  style='display:inline;' bgcolor='#[num2hex(r_eyes, 2)][num2hex(g_eyes, 2)][num2hex(b_eyes)]'><tr><td>__</td></tr></table></font><br>"
 
-	var/datum/species/current_species = GLOB.all_species[species]
+	var/datum/species/current_species = GLOB.roundstart_species[species]
```
Comment (SplinterGP) [inline code/modules/client/preferences.dm:255]: Does not need to be roundstart_species.

File: code/modules/client/preferences.dm [hunk 4]
```diff
@@ -667,7 +667,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 			real_name = newname
 
 		if("randomize_name")
-			var/datum/species/S = GLOB.all_species[species]
+			var/datum/species/S = GLOB.roundstart_species[species]
```
Comment (SplinterGP) [inline code/modules/client/preferences.dm:670]: does not need to be roundstart_species

File: code/modules/client/preferences_savefile.dm [hunk 5]
```diff
@@ -362,7 +362,7 @@
 	random_name		= sanitize_integer(random_name, FALSE, TRUE, initial(random_name))
 	gender			= sanitize_gender(gender)
 	age				= sanitize_integer(age, AGE_MIN, AGE_MAX, initial(age))
-	species			= sanitize_inlist(species, GLOB.all_species, initial(species))
+	species			= sanitize_inlist(species, GLOB.roundstart_species, initial(species))
```
Comment (SplinterGP) [inline code/modules/client/preferences_savefile.dm:365]: leave it as all_species.

File: code/modules/client/preferences_savefile.dm [hunk 6]
```diff
@@ -440,7 +440,7 @@
...
```
Comment (SplinterGP) [inline code/modules/client/preferences_savefile.dm:443]: Ditto, leave as all_species

File: code/modules/mob/new_player/new_player.dm [hunk 7]
```diff
@@ -190,10 +190,10 @@
 
 				observer.timeofdeath = world.time
 
-				var/datum/species/species = GLOB.all_species[client.prefs.species] || GLOB.all_species[DEFAULT_SPECIES]
+				var/datum/species/species = GLOB.roundstart_species[client.prefs.species] || GLOB.roundstart_species[DEFAULT_SPECIES]
```
Comment (SplinterGP) [inline code/modules/mob/new_player/new_player.dm:193]: leave as all_species, like the others you want to look in all species, only when choosing you should show roundstart_species as option or when defaulting it.

File: code/modules/mob/new_player/new_player.dm [hunk 8]
```diff
@@ -359,7 +359,7 @@
 /mob/new_player/get_species()
 	var/datum/species/chosen_species
 	if(client.prefs.species)
-		chosen_species = GLOB.all_species[client.prefs.species]
+		chosen_species = GLOB.roundstart_species[client.prefs.species]
```
Comment (SplinterGP) [inline code/modules/mob/new_player/new_player.dm:362]: Keep as all_species

File: code/__HELPERS/global_lists.dm (hunk 1 above)
Comment (SplinterGP) [reply code/__HELPERS/global_lists.dm:57]: otherwise this is just putting all the species in it like all_species.

File: code/__HELPERS/mobs.dm [hunk 9]
```diff
@@ -44,7 +44,7 @@
 
 
 /proc/get_playable_species()
-	var/list/playable_species = list(GLOB.all_species[DEFAULT_SPECIES])
+	var/list/playable_species = list(GLOB.roundstart_species[DEFAULT_SPECIES])
```
Comment (SplinterGP) [inline code/__HELPERS/mobs.dm:47]:
```suggestion
	var/list/playable_species = list(GLOB.roundstart_species)
```

File: code/modules/mob/new_player/preferences_setup.dm [hunk 10]
```diff
@@ -208,7 +208,7 @@
 
 /datum/preferences/proc/random_character()
 	gender = pick(MALE, FEMALE)
-	var/datum/species/S = GLOB.all_species[DEFAULT_SPECIES]
+	var/datum/species/S = GLOB.roundstart_species[DEFAULT_SPECIES]
```
Comment (SplinterGP) [inline code/modules/mob/new_player/preferences_setup.dm:211]: this makes a random character, meaning this should be what the datum/preferences species should be, so it picks the species you are
```suggestion
	var/datum/species/S = GLOB.all_species[species]
```

**Review by SplinterGP — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/species.dm [hunk 11]
```diff
@@ -329,6 +332,7 @@
 	brute_mod = 1.05
 	burn_mod = 1.05
 	slowdown = 1.05
+	joinable_roundstart = FALSE
```
Comment (HolisticDelight) [inline code/modules/mob/living/carbon/human/species.dm:335]: Why you cant use vat grown as your species? I just saw many peoples enjoyed having vat grown chars.

Comment (XSlayer300) [reply code/modules/mob/living/carbon/human/species.dm:335]: Vat-growns and vatborns are entirely two different things.

Vat-growns are basically cloned guys during the round, and they can take slightly less damage.
Vatborns are also cloned guys, but they have matured well enough to survive like a human, performing everything like a human could.

**Comment by XSlayer300** [comment]
shush

**Comment by XSlayer300** [comment]
PR broken as of latest update.

**Comment by XSlayer300** [comment]
For some reason, spawning as a Synthethic with the Vatborn species will result in the Vatborn overwriting the species of the synth itself, rather than the other way around.

**Comment by XSlayer300** [comment]
All bugs at this point are fixed, so it should be ready.

**Review by SplinterGP — APPROVED** [review]
the code can be better generally when deciding how to spawn and all, but as that goes its a refactor and dont need done, there we go.

**Review by TiviPlus — APPROVED** [review]

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-27). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Pull request merged**